proving/proving_options.go

```diff
@@ -49,3 +58,27 @@ func WithDataSource(cfg config.Config, nodeId, commitmentAtxId []byte, datadir s
 		return nil
 	}
 }
+
+func defaultOpts() *option {
```

Comment:
I think this is only used in a single place and can therefore be inlined.

Reply:
What's wrong with defining the defaults next to structure definition? I think it makes it easier to find.